### PR TITLE
feat: Support for team/hierarchies overrides

### PR DESCRIPTION
**Pull request by nikhilunni**
## Change description

Allow Backstage instances to override the default Cortex team / team hierarchy mappings.
To use, we provide a new function to the `ExtensionApi`:
```
getTeamOverrides?(entities: Entity[]): Promise<TeamOverrides>
```
Additionally, this will allow you to add team metadata you would not be able to get from the default mapping alone (descriptions, quick links, slack channels for notifications, to name a few).

## Type of change

- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Improvement (improves codebase without changing functionality)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,12 +47,10 @@
     "@backstage/core-plugin-api": "^1.0.6",
     "@backstage/plugin-catalog-react": "^1.1.4",
     "@backstage/theme": "^0.2.16",
-    "@cortexapps/backstage-plugin-extensions": "^0.0.11",
+    "@cortexapps/backstage-plugin-extensions": "0.0.12",
     "@material-ui/core": "^4.11.0",
     "@material-ui/icons": "^4.11.2",
     "@material-ui/lab": "4.0.0-alpha.45",
-    "@mui/icons-material": "^5.0.4",
-    "@mui/material": "^5.0.4",
```
Comment (FoodProduct) [inline package.json:55]: This kind of feels like a reversion 🙃. [MUI is material-ui v5](https://mui.com/blog/material-ui-is-now-mui/) (at some point we really should fully upgrade).

File: src/components/SettingsPage/SettingsPage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import { CatalogApi, catalogApiRef } from "@backstage/plugin-catalog-react";
+import { Fixtures } from "../../utils/TestUtils";
+import { renderWrapped } from "../../utils/TestUtils";
+import { SettingsPage } from "./SettingsPage";
+import { captor, mock } from 'jest-mock-extended';
+import { ExtensionApi } from "@cortexapps/backstage-plugin-extensions";
+import { extensionApiRef } from "../../api/ExtensionApi";
+import { waitFor } from "@testing-library/react";
+
+describe('<SettingsPage/>', () => {
```
Comment (FoodProduct) [inline src/components/SettingsPage/SettingsPage.test.tsx:27]: Thanks for adding tests! 🙌

File: package.json (hunk 1 above)
Comment (nikhilunni) [reply package.json:55]: yupp agreed, we just have a dependency we're not using at this point 🤷🏽 material-ui -> mui should be done as its own change

Comment (nikhilunni) [reply package.json:55]: also backstage itself is still on `material-ui@4.x`

Comment (FoodProduct) [reply package.json:55]: > also backstage itself is still on `material-ui@4.x`

Oof good find! 😆

**Pull request merged**